lib/source/common.js

```diff
@@ -0,0 +1,184 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const DEFAULT_INTERVAL = 60000;
+
+const deepDiff = require('deep-diff').diff;
+
+const Source = (Parser) => class extends EventEmitter {
```

Comment:
Shamelessly boosted that idiom from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Mix-ins